internal/mempool/v1/tx.go

```diff
@@ -198,3 +199,50 @@ func (txs *TxStore) GetOrSetPeerByTxHash(hash [mempool.TxKeySize]byte, peerID ui
 	wtx.peers[peerID] = struct{}{}
 	return wtx, false
 }
+
+// WrappedTxList implements a thread-safe list of *WrappedTx objects that can be
+// used to build generic transaction indexes in the mempool.
```

Comment:
I would pass a comparator function into the constructor of this type and cached so that you could make the caller oblivious to the implementation of of the ordered list (which sort of functions like a heap.) 

Also I think the fact that you get the index where you need to insert at, under a readlock, and then do the insert in a write lock means that you could attempt to insert more than one item at the same index which will probably result in out of order storage.

Reply:
I don't think this change is necessary. I will collapse `InsertAt` and `GetSortedIndex` into just a single `Insert` method.

Reply:
Actually, that would remove some code from the mempool logic. I'll look into adding it.